lib/streamlit/elements/image.py

```diff
@@ -386,6 +387,42 @@ def image_to_url(
         return ""
 
 
+def _get_svg_width_and_height(image: str) -> Tuple[str, str]:
+    root_idx = image.find(">")
+    root = image[:root_idx]
+    width = "100"
```

Comment:
Is 100x100 a reasonable default size? I also could imagine that it might break some existing uses of SVG.

Reply:
You're right 🥇 Thank you for noticing this, setting any arbitrary default height and width on any SVG will break the user space 👍 

The only case that it makes sense to set it, is:
1. The SVG does not have property width set
2. We can read width from the viewBox attribute

The `width` on SVG is required to allow Streamlit scale the image, if it's not set the image won't be scaled. 
For the above reasons, I've added the function ` _check_if_svg_can_be_rendered_as_img` and the width will be set only when conditions are met.

```
def _check_if_svg_can_be_rendered_as_img(image: str) -> bool:
    if image.find("xmlns:xlink") != -1:
        return False
    if image.find("width") == -1 and not _get_svg_width(image):
        return False
    return True
```

When I'm reading about SVGs from [the article](https://css-tricks.com/scale-svg/#aa-the-svg-scaling-toolbox) linked by you, which is very helpful, the `viewBox` 3d and 4th parameters actually define image `width` and `height`, if setting them explicitly breaks the user space it's because we've improperly rendered them previously and after this change it will work as intended.

Reply:
So, if an SVG has a `viewbox` but doesn't have `width` property set, it won't be scaled correctly with our `img` tag setup? The [article](https://css-tricks.com/scale-svg/#aa-option-1-use-image-auto-sizing) seems to suggest that it somehow works. Maybe we need to tweak our CSS props on the frontend?